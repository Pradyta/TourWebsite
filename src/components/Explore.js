import React from "react";
import "../style.css";
const Explore = ({ title, placesAPI }) => {
  return (
    <>
      <div id="explore-1" className="relative my-10 md:mt-10">
        <div className="travigo-container">
          <div className="flex items-center justify-center text-center mb-11 md:mb-17">
            <h1 className="text-5xl lg:text-4xl md:text-3xl sm:text-2xl xsm:text-xl font-bold filter drop-shadow-lg text-slate-900">
              {title}
            </h1>
          </div>
          <div className="grid items-center grid-cols-4 lg:grid-cols-3 md:grid-cols-2 gap-6">
            {placesAPI?.map((val, i) => (
              <div
                key={i}
                className="item flex items-center gap-5 sm:gap-3 rounded-lg hover:bg-emerald-300 transition-all duration-300 cursor-pointer hover:scale-105"
              >
                <div className="corner-image flex items-center">
                  <img
                    src={val.placeImg}
                    alt={val.location}
                    className="w-20 h-20 sm:w-16 sm:h-16 rounded-lg filter drop-shadow-lg"
                  />
                </div>
                <div className="flex items-start flex-col text-slate-900">
                  <h1 className="text-lg sm:text-sm font-bold">
                    {val.location}
                  </h1>

                  <p className="font-normal lg:text-sm text-base sm:text-xs">
                    {val.distance}
                  </p>
                </div>

                <a href={val.link} target="_blank">
                  <button className="bg-white-500 hover:bg-grey-700 text-black font-bold py-2 px-4 rounded-full">
                    Explore Now
                  </button>
                </a>
              </div>
            ))}
          </div>
        </div>
      </div>
    </>
  );
};

export default Explore;
