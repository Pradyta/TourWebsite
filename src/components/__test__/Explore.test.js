import { render, screen } from '@testing-library/react';
import Explore from '../Explore';


describe('Explore Component', () => {
    test('should render Explore compoennt', () => {
    const {asFragment} = render( 
        <Explore />,
    );
    expect(asFragment(<Explore/>)).toMatchSnapshot();
    });

    test('test', () => {
        render(<Explore/>);
        expect(true).toBe(true);
    });
});