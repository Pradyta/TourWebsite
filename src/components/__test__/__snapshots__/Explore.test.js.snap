// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Component should render Explore compoennt 1`] = `
<DocumentFragment>
  <div
    class="relative my-7 md:mt-3"
    id="explore-1"
  >
    <div
      class="travigo-container"
    >
      <div
        class="flex items-center justify-center text-center mb-11 md:mb-7"
      >
        <h1
          class="text-5xl lg:text-4xl md:text-3xl sm:text-2xl xsm:text-xl font-bold filter drop-shadow-lg text-slate-900"
        />
      </div>
      <div
        class="grid items-center grid-cols-4 lg:grid-cols-3 md:grid-cols-2 gap-5"
      />
    </div>
  </div>
</DocumentFragment>
`;
